[BUGFIX] missing modules docs for tryInvoke, compare, isEqual #16079

jenweber opened the pull request:
See https://github.com/emberjs/ember.js/issues/16079. This should target at least 2.18 LTS candidate and 3.0-beta. 2.16 LTS would be good too. I have no idea how any of that is done so let me know if I need to open more PRs against other branches.

Needs a good look by @toddjordan 

Thank you to sukima for reporting!

jenweber committed:
[BUGFIX] modules docs for tryInvoke, compare, isEqual #16079

jenweber commented:
The test suite passes locally.

CI fails at yarn install.
```error An unexpected error occurred: "https://registry.yarnpkg.com/pinkie-promise/-/pinkie-promise-2.0.1.tgz: Request failed \"404 Not Found\"".```

jenweber commented:
Also missing `compare` which should belong to `@ember/utils`

[docs here](https://github.com/emberjs/ember.js/blob/5f4ac602813d8fbd78f26fc955acf693e0b3d62a/packages/ember-runtime/lib/mixins/comparable.js)

toddjordan commented on packages/ember-runtime/lib/is-equal.js:
````diff
@@ -30,7 +33,7 @@
   ```
 
   @method isEqual
````
Also needs an `@static` to show up as a function

jenweber replied:
resolved

toddjordan commented on packages/ember-runtime/lib/compare.js:
````diff
@@ -72,7 +76,7 @@ function spaceship(a, b) {
   ```
 
  @method compare
````
needs an `@static`

jenweber replied:
resolved

toddjordan commented on packages/ember-utils/lib/invoke.js:
````diff
@@ -12,7 +15,7 @@ import applyStr from './apply-str';
   ```
 
   @method canInvoke
````
`@static`

jenweber replied:
I removed canInvoke - turns out it's not a package/module which makes sense because it is private API

toddjordan commented on packages/ember-utils/lib/invoke.js:
````diff
@@ -35,7 +38,7 @@ export function canInvoke(obj, methodName) {
   ```
 
   @method tryInvoke
````
`@static`

jenweber replied:
resolved

jenweber commented:
Thanks, I'll get those changes made

jenweber commented:
@toddjordan I added static to everything and double-checked that these methods are actually available in the utils module/package. `canInvoke` wasn't, so I reverted those changes.

toddjordan commented:
Tried it out.  Looks good!  Thanks @jenweber !!!
